src/domain.jl

```diff
@@ -82,10 +82,6 @@ function affect!(integrator, f::AbstractDomainAffect{T,S,uType}) where {T,S,uTyp
         dtcache = integrator.dt
         integrator.dt *= scalefactor
         dt_modified = true
-
-        # adjust new time step to bounds and time stops
-        fix_dt_at_bounds!(integrator)
-        modify_dt_for_tstops!(integrator)
```

Comment:
This should be fine - these two cases (`dt` leads to a time point `> tspan[2]` or `> top(integrator.tstops)`) are handled before the for loop, and hence they can't occur anymore if `dt` is decreased in the for loop.